image/icns.ksy

```diff
@@ -0,0 +1,889 @@
+            type: u4
+            doc: The height of the icon in pixels.
+        seq:
+          - id: icon
+            size: width * height / 8
+            doc: |
+              The icon bitmap data,
+              as a packed sequence of 1-bit pixels,
+              in MSB-first bit order
+              (i. e. pixels that come earlier in the file go into the higher bits).
+              0 is white and 1 is black.
+          - id: mask
+            size: width * height / 8
+            doc: |
+              The icon mask data,
+              as a packed sequence of 1-bit pixels,
+              in MSB-first bit order
+              (i. e. pixels that come earlier in the file go into the higher bits).
+              0 is transparent and 1 is solid.
+      icon_x4_data:
+        doc: |
+          The data for a 4-bit color bitmap icon.
+          These icons do not contain a mask,
+          but use the mask from the 1-bit monochrome icon of the same size from the same family.
+        params:
+          - id: width
+            type: u4
+            doc: The width of the icon in pixels.
+          - id: height
+            type: u4
+            doc: The height of the icon in pixels.
+        seq:
+          - id: icon
+            size: width * height / 2
+            doc: |
+              The icon bitmap data,
+              as a packed sequence of 4-bit pixels,
+              in MSB-first bit order
+              (i. e. pixels that come earlier in the file go into the higher bits).
+              Color values come from the Mac OS default 4-bit color palette.
+      icon_x8_data:
+        doc: |
+          The data for an 8-bit color bitmap icon.
+          These icons do not contain a mask,
+          but use the mask from the 1-bit monochrome icon of the same size from the same family.
+        params:
+          - id: width
+            type: u4
+            doc: The width of the icon in pixels.
+          - id: height
+            type: u4
+            doc: The height of the icon in pixels.
+        seq:
+          - id: icon
+            size: width * height
+            doc: |
+              The icon bitmap data,
+              as a packed sequence of 8-bit pixels.
+              Color values come from the Mac OS default 8-bit color palette.
+      icon_rgb_data:
+        doc: |
+          The data for a 24-bit RGB bitmap icon.
+          These icons do not contain a mask -
+          the mask is stored as a separate element in the same icon family.
+        params:
+          - id: width
+            type: u4
+            doc: The width of the icon in pixels.
+          - id: height
+            type: u4
+            doc: The height of the icon in pixels.
+        seq:
+          - id: data
+            type:
+              switch-on: _io.size == width * height * 3
+              cases:
+                false: icns_style_packbits
+                # If true, the data is uncompressed and should be read as a raw byte array.
```

Comment:
This attribute doesn't have the `size` key, so it won't be read as a byte array. If the condition `_io.size == width * height * 3` is `true`, the value of `data` will be `null` instead (or the equivalent in the particular target language).

Reply:
Thanks, I completely missed that.

I think I already mentioned this in my previous PR, but I'm *really* not a fan of how the implicit default case in `switch-on`s works, and that it's *necessary* to use the implicit default case to have a byte array case in a `switch-on`. I've explained my thoughts on this in more detail in a separate issue: kaitai-io/kaitai_struct#799